docs/advanced-guide/injecting-databases-drivers/page.md

```diff
@@ -828,3 +828,139 @@ func getUser(c *gofr.Context) (any, error) {
 	return user, nil
 }
 ```
+## SurrealDB
+
+GoFr supports injecting SurrealDB database that supports the following interface. Any driver that implements the interface can be added
+using `app.AddSurrealDB()` method, and users can use Surreal DB across application through the `gofr.Context`.
+
+```go
+// SurrealDB defines an interface representing a SurrealDB client with common database operations.
+type SurrealDB interface {
+
+// Query executes a Surreal query with the provided variables and returns the query results as a slice of interfaces{}.
+// It returns an error if the query execution fails.
+Query(ctx context.Context, query string, vars map[string]any) ([]any, error)
+
+// Create inserts a new record into the specified table and returns the created record as a map.
+// It returns an error if the operation fails.
+Create(ctx context.Context, table string, data any) (map[string]any, error)
+
+// Update modifies an existing record in the specified table by its ID with the provided data.
+// It returns the updated record as an interface and an error if the operation fails.
+Update(ctx context.Context, table string, id string, data any) (any, error)
+
+// Delete removes a record from the specified table by its ID.
+// It returns the result of the delete operation as an interface and an error if the operation fails.
+Delete(ctx context.Context, table string, id string) (any, error)
+
+// Select retrieves all records from the specified table.
+// It returns a slice of maps representing the records and an error if the operation fails.
+Select(ctx context.Context, table string) ([]map[string]any, error)
+
+HealthChecker
+}
+
+// SurrealBDProvider is an interface that extends SurrealDB with additional methods for logging, metrics, or connection management.
+// It is typically used for initializing and managing SurrealDB-based data sources.
+type SurrealBDProvider interface {
+SurrealDB
+
+provider
+}
+
```

Comment:
The formatting is not good.

Copy paste it to a go file, format it, then copy paste it back here

Reply:
Thanks 

Side note for @hougesen, there is a need for a GitHub action

- https://github.com/hougesen/mdsf/issues/633